.buildkite/scripts/common.sh

```diff
@@ -336,19 +399,89 @@ config_git() {
   fi
 }
 
+defineModuleFromTheChangeSet() {
+  # This method gathers the module name, if required, in order to run the ITs only if the changeset affects a specific module.
+  # For such, it's required to look for changes under the module folder and exclude anything else such as asciidoc and png files.
+  # This method defines and exports the MODULE variable with a particular module name or '' if changeset doesn't affect a specific module
+  local module_path=$1
+  if [[ "$module_path" == *"x-pack/"* ]]; then
+    local pattern=("$XPACK_MODULE_PATTERN")
+  else
+    local pattern=("$OSS_MODULE_PATTERN")
+  fi
+  local module_name="${module_path#*module/}"
+  local module_path_transformed=$(echo "$module_path" | sed 's/\//\\\//g')
+  local module_path_exclussion="((?!^${module_path_transformed}\\/).)*\$"
+  local exclude=("^(${module_path_transformed}|((?!\\/module\\/).)*\$|.*\\.asciidoc|.*\\.png)")
+  if are_paths_changed "${pattern[@]}" && ! are_changed_only_paths "${exclude[@]}"; then
+    export MODULE=${module_name}
+  elif [ -d "${module_path}" ]; then
+    export MODULE="aws"                 # TODO: remove this line and uncomment the line below when the issue https://github.com/elastic/ingest-dev/issues/2993 is solved
+    # export MODULE=''                  # TODO: uncomment the line when the issue https://github.com/elastic/ingest-dev/issues/2993 is solved
+  fi
+}
+
+terraformInit() {
+  local dir=$1
+  echo "Terraform Init on $dir"
+  pushd "${dir}" > /dev/null
+  terraform init
+  popd > /dev/null
+}
+
+withAWS() {
+  # This method gathers the masked AWS credentials from pre-command hook and sets the right AWS variable names.
+  export AWS_ACCESS_KEY_ID=$BEATS_AWS_ACCESS_KEY
+  export AWS_SECRET_ACCESS_KEY=$BEATS_AWS_SECRET_KEY
+  export TEST_TAGS="${TEST_TAGS:+$TEST_TAGS,}aws"
+}
+
+startCloudTestEnv() {
+  local dir=$1
+  withAWS
+  echo "--- Run docker-compose services for emulated cloud env"
+  docker-compose -f .ci/jobs/docker-compose.yml up -d                     #TODO: move all docker-compose files from the .ci to .buildkite folder before switching to BK
+  with_Terraform
+  terraformInit "$dir"
+  export TF_VAR_BRANCH=$(echo "${BUILDKITE_BRANCH}" | tr '[:upper:]' '[:lower:]' | sed 's/[^a-z0-9-]/-/g')
+  export TF_VAR_BUILD_ID="${BUILDKITE_BUILD_ID}"
+  export TF_VAR_CREATED_DATE=$(date +%s)
+  export TF_VAR_ENVIRONMENT="ci"
+  export TF_VAR_REPO="${REPO}"
+  pushd "${dir}" > /dev/null
+  terraform apply -auto-approve
+  popd > /dev/null
+}
+
+cleanup() {
+  echo "---Terraform Cleanup"
+  .ci/scripts/terraform-cleanup.sh "${MODULE_DIR}"              #TODO: move all docker-compose files from the .ci to .buildkite folder before switching to BK
```

Comment:
I think, there is a missing context what's the reason for running this terraform clean up function. A comment can help in the future. 
In addition, should this particular script run only if the stage `cloud` ran for the `x-pack/metricbeat`? Or will it work regardless of the stages in all the beats in case they call the cleanup function?

Reply:
I've got what you mean, In this case I would suggest using the divided "cleanup" step with Slack notification or pipeline with email notification. @v1v WDYT?

Reply:
To illustrate a bit more, the current implementation in Jenkins does not fail when tearing down the resources with terraform/docker:

- https://github.com/elastic/beats/blob/3f462221002e595c5763bc626f1dc1a8e1b508e2/Jenkinsfile#L945-L950

It uses `returnStatus: true` that means do nothing, see 

> returnStatus : boolean (optional)
Normally, a script which exits with a nonzero status code will cause the step to fail with an exception. If this option is checked, the return value of the step will instead be the status code. You may then compare it to zero, for example.

See https://www.jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#sh-shell-script


You can get the same:
either you can `|| true` or use your additional cleanup with [soft_fail](https://buildkite.com/changelog/56-command-steps-can-now-be-made-to-soft-fail) in Buildkite.

> WDYT?

Sounds good to me. But let me ask @dliappis about his preference.